train.c

```diff
@@ -58,13 +103,36 @@ static int get_action_epsilon_greedy(char *table, td_agent_t *agent)
     epsilon *= decay_factor;
     return act;
 }
+#endif
+
+static float update_state_value(int after_state_hash,
+                                float next_target,
+                                rl_agent_t *agent)
+{
+#if EPISODE_REWARD
+    float target = -GAMMA * next_target;
```

Comment:
The term `target` is for TD learning, I believe it might lead to confusion if we reuse the same term in Monte Carlo Search.(In Monte Carlo, we might care about `total return`?) Perhaps you could split the update functions for these two algorithms into different functions to make the code more readable.

Reply:
Yes, It is "return" in MC, or "TD target" in TD, is `new_value` a better name?